### DISCUSSION TITLE
About the calculation of `katz_centrality_numpy`

### DISCUSSION THREAD

**Discussion by waving7799** [networkx/networkx#5375] [Q&A]
I recently noticed that when I call  `katz_centrality_numpy` to extract katz centrality for a graph, the nodes without any edges have the centrality values, which is not supposed to be according to the definition of katz centrality. 

I wondered if there are some problems with the calculation of katz centrality. When solving the  linear matrix equation, it seems that the dependent variable b should not be all 1. According to the reference *A New Status Index Derived from Sociometric Index.*， maybe the values of b should be the degree of each node?

**Comment by dschult** [comment]
The definition of Katz Centrality has been around since 1953. Can you compare your notation with [the Wikipedia description of Katz Centrality](https://en.wikipedia.org/wiki/Katz_centrality) to figure out what the difference are.  We are unlikely to change from the standard definition. Perhaps this reference you cite (can you "update" your post to include the reference?) has a new definition that should be named something different. On the other hand, if the NetworkX code does not match the standard definition we should obvious correct it.  Perhaps you can compare your results for `katz_centrality_numpy` with `katz_centrality`.
